hey-discount/src/main/webapp/resources/js/script.js

```diff
@@ -0,0 +1,878 @@
+        document.getElementById("back2").style.display = "block";
+        document.getElementById("window2").style.display = "block";
+    }
+
+    cancelSignUp(){
+        document.getElementById("back2").style.display = "none";
+        document.getElementById("window2").style.display = "none";
+    }
+
+    authorize(event) {
+        var item = event.target;
+        if (item.login.value) {
+            this.user = item.login.value;
+            this.view.user = item.login.value;
+            item.login.value = '';
+        }
+        if (item.radio1.checked) {
+            item.radio1.checked = false;
+            item.radio2.checked = false;
+            this.isVendor = false;
+        }
+        if (item.radio2.checked) {
+            item.radio2.checked = false;
+            item.radio1.checked = false;
+            this.isVendor = true;
+        }
+        if (item.password.value) {
+            item.password.value = '';
+        }
+        this.cancelSignUp();
+        this.buttonSigningOut = document.querySelector(".header-button");
+        this.buttonSigningOut.addEventListener("click", () => this.signOut());
+        document.getElementById("container").innerHTML = "";
+        this.countOfPostsOnFeed = 0;
+        this.loadMoreAds();
+    }
+
+    publishAd(event) {
+        var departure = 'begin';
+        var ad = {
+            id: (this.model.size() + 1).toString(),
+            title: '',
+            description: '',
+            createdAt: new Date(),
+            link: '',
+            vendor: this.user,
+            photoLink: '',
+            hashTags: null,
+            discount: '',
+            validUntil: null,
+            rating: 0,
+            reviews: []
+        };
+        var item = event.target;
+        if (item.adTitle.value) {
+            ad.title = item.adTitle.value;
+            item.adTitle.value = '';
+        }
+        if (item.shortDescription.value) {
+            ad.description = item.shortDescription.value;
+            item.shortDescription.value = '';
+        }
+        if (item.link.value) {
+            ad.link = item.link.value;
+            item.link.value = '';
+        }
+        if (item.photoLink.value) {
+            ad.photoLink = item.photoLink.value;
+            item.photoLink.value = '';
+        }
+        if (item.formHashtags.value) {
+            ad.hashTags = item.formHashtags.value.split(' ');
+            item.formHashtags.value = '';
+        }
+        if (item.validityPeriod.valueAsDate) {
+            ad.validUntil = item.validityPeriod.valueAsDate;
+            item.validityPeriod.valueAsDate = null;
+        }
+        if (item.discountAmount.value) {
+            ad.discount = item.discountAmount.value;
+            item.discountAmount.value = '';
+        }
+        if (this.adForEdit) {
+            this.view.edit(this.adForEdit.id, this.model.edit(this.adForEdit.id, ad));
+        } else {
+            this.view.addItem(this.model.add(ad), departure);
```

Comment:
Instead of 'departure' you could just use boolean value, for example `appendToEnd` with default value `true`, here just pass `false` as a parameter, on line 602 omit it 🙂